pyopencl/__init__.py

```diff
@@ -508,8 +508,16 @@ def build(self, options=None, devices=None, cache_dir=None):
             cache_dir = getattr(self._context, "cache_dir", None)
 
         build_descr = None
-        if _PYOPENCL_NO_CACHE and self._prg is None:
-            build_descr = "uncached source build (cache disabled by user)"
+        no_cache = _PYOPENCL_NO_CACHE
+        # Turn off caching when using pocl to avoid extra compile from
+        # get_info(BINARIES). See https://github.com/inducer/pyopencl/issues/731.
+        from pyopencl.characterize import get_pocl_version
```

Comment:
Maybe add a `characterize` function on whether there is an effective source -> executable cache? (Specified to return `None` for "unknown", `False` for no, and `True` for yes.)